Projects/Addons/Sources/Views/AddonSelectSubOptionScreen.swift

```diff
@@ -49,12 +46,20 @@ struct AddonSelectSubOptionScreen: View {
                 }
             }
             .padding(.top, .padding16)
+            .onAppear {
+                if let preSelectedQuote = changeAddonNavigationVm.changeAddonVm!.selectedQuote {
+                    self.selectedQuote = preSelectedQuote
+                } else if let firstQuote = addonOffer.quotes.first {
+                    self.selectedQuote = firstQuote
+                }
+            }
         }
         .hDisableScroll
         .hFormAttachToBottom {
             hSection {
                 VStack(spacing: .padding8) {
                     hButton.LargeButton(type: .primary) {
+                        changeAddonNavigationVm.changeAddonVm?.selectedQuote = selectedQuote
                         changeAddonNavigationVm.isChangeCoverageDaysPresented = nil
```

Comment:
This could be done more elegant with having 
`@EnvironmentObject var router: Router `in the beggining of the AddonSelectSubOptionScreen view
and just do `router.dimiss()` here

Reply:
Same in one bellow.